### PR TITLE
Add local-only isVisibleInSecondaryCamera flag to entities

### PR DESCRIPTION
**Pull request by SamGondelman**
(shouldn't require a protocol bump)

Adds a local flag to all entity types that let's you disable their rendering in the secondary camera view while continuing to render them in the main view.

https://highfidelity.manuscript.com/f/cases/12763/camera-flicker-when-flying-was-Spectator-Cam-issues-from-Phlash

Test plan:
- Stand in front of a mirror.
- Run [this](https://gist.githubusercontent.com/SamGondelman/a44a950be3d513933c0e7ec70e01c556/raw/64774f1ec9c5cf9ad5c93d09ebce6a3beb981f6d/VisibleInSecondaryCameraTest.js).
- You should see a white box, two hats, and a green wireframe box, in both the main view and mirror.
- Press 'u'.  They should all disappear in the main view and the mirror.  Press it again and they should come back.
- Press 'i'.  They should disappear in the mirror, but remain in the main view.  Press it again and they should come back in the mirror.
- In first person, you should be able to see your head in the mirror.  When you walk, your eyes shouldn't appear in your view.  In third person, your head should appear in the main view and the mirror.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10070/HighFidelity-Beta-PR13227-e6741a363ec419afead6949922bb0eb18ef5eab8.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10070/HighFidelity-Beta-PR13227-e6741a363ec419afead6949922bb0eb18ef5eab8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10070/HighFidelity-Beta-Interface-PR13227-e6741a363ec419afead6949922bb0eb18ef5eab8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10070/HighFidelity-Beta-PR13227-e6741a363ec419afead6949922bb0eb18ef5eab8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10070/HighFidelity-Beta-Interface-PR13227-e6741a363ec419afead6949922bb0eb18ef5eab8.exe

**Comment by AndrewMeadows** [comment]
What was the motivating bug report for this?  I'm curious why some entities should NOT be visible in the secondary camera.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10292/HighFidelity-Beta-PR13227-2c36d30.apk

File: libraries/entities/src/EntityItem.cpp [hunk 1]
```diff
@@ -2760,6 +2761,28 @@ void EntityItem::setVisible(bool value) {
     }
 }
 
+bool EntityItem::isVisibleInSecondaryCamera() const {
+    bool result;
+    withReadLock([&] {
+        result = _isVisibleInSecondaryCamera;
+    });
+    return result;
+}
```
[the remark this answers is not in the document]
Comment (SamGondelman) [reply libraries/entities/src/EntityItem.cpp]: Agreed, but deferring for now.

File: libraries/entities/src/EntityPropertyFlags.h [hunk 2]
```diff
@@ -249,6 +249,8 @@ enum EntityPropertyList {
     PROP_MATERIAL_MAPPING_ROT,
     PROP_MATERIAL_DATA,
 
+    PROP_VISIBLE_IN_SECONDARY_CAMERA, // not sent over the wire, only used locally
```
[the remark this answers is not in the document]
Comment (SamGondelman) [reply libraries/entities/src/EntityPropertyFlags.h]: Done.

**Comment by SamGondelman** [comment]
@AndrewMeadows @samcake wanted this to fix the issue of the spectator cam clipping into the camera model.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10292/HighFidelity-Beta-PR13227-2c36d30.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10292/HighFidelity-Beta-Interface-PR13227-2c36d30.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10292/HighFidelity-Beta-PR13227-2c36d30.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10292/HighFidelity-Beta-Interface-PR13227-2c36d30.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10301/HighFidelity-Beta-PR13227-86ee5c1.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10301/HighFidelity-Beta-PR13227-86ee5c1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10301/HighFidelity-Beta-Interface-PR13227-86ee5c1.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Items appear and reappear in the mirror as expected.

**Pull request merged**